src/main/java/seedu/ccacommander/logic/commands/EditEnrolmentCommand.java

```diff
@@ -0,0 +1,236 @@
+        requireNonNull(memberIndex);
+        requireNonNull(eventIndex);
+        requireNonNull(editEnrolmentDescriptor);
+
+        this.memberIndex = memberIndex;
+        this.eventIndex = eventIndex;
+        this.editEnrolmentDescriptor = new EditEnrolmentDescriptor(editEnrolmentDescriptor);
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Member> lastShownMemberList = model.getFilteredMemberList();
+        List<Event> lastShownEventList = model.getFilteredEventList();
+
+        if (memberIndex.getZeroBased() >= lastShownMemberList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_MEMBER_DISPLAYED_INDEX);
+        }
+        if (eventIndex.getZeroBased() >= lastShownEventList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_EVENT_DISPLAYED_INDEX);
+        }
+
+        Member targetMember = lastShownMemberList.get(memberIndex.getZeroBased());
+        Event targetEvent = lastShownEventList.get(eventIndex.getZeroBased());
+        Enrolment enrolmentToCheck =
+                new Enrolment(targetMember.getName(), targetEvent.getName(), new Hours("0"), new Remark("HOLDER"));
+
+        model.updateFilteredEnrolmentList(new EnrolmentExistsPredicate(enrolmentToCheck));
+        List<Enrolment> enrolmentToEditList = model.getFilteredEnrolmentList();
+        if (enrolmentToEditList.size() == 0) {
+            throw new CommandException(String.format(Messages.MESSAGE_ENROLMENT_DOES_NOT_EXIST,
+                    memberIndex.getOneBased(), eventIndex.getOneBased()));
+        }
+        assert enrolmentToEditList.size() == 1 : "There should not be duplicate enrolments";
+
+        Enrolment enrolmentToEdit = enrolmentToEditList.get(0);
+        Enrolment editedEnrolment = createEditedEnrolment(enrolmentToEdit, editEnrolmentDescriptor);
+
+        model.setEnrolment(enrolmentToEdit, editedEnrolment);
+        model.updateFilteredEnrolmentList(PREDICATE_SHOW_ALL_ENROLMENTS);
+        model.commit(String.format(MESSAGE_COMMIT, editedEnrolment.getMemberAndEventEnrolment()));
+        return new CommandResult(String.format(MESSAGE_EDIT_ENROLMENT_SUCCESS, Messages.format(editedEnrolment)));
+    }
+
+    /**
+     * Creates and returns a {@code Enrolment} with the details of {@code enrolmentToEdit}
+     * edited with {@code editEnrolmentDescriptor}.
+     */
+    private static Enrolment createEditedEnrolment(Enrolment enrolmentToEdit,
+                                                    EditEnrolmentDescriptor editEnrolmentDescriptor) {
+        assert enrolmentToEdit != null;
+
+        Name memberName = editEnrolmentDescriptor.getMemberName().orElse(enrolmentToEdit.getMemberName());
+        Name eventName = editEnrolmentDescriptor.getEventName().orElse(enrolmentToEdit.getEventName());
```

Comment:
small nit but since these two fields are unmodifiable through editEnrolment, I think can just put enrolmentToEdit.getMemberName() for conciseness :0 Its not too significant though